### ISSUE TITLE
Use .NET `TimeProvider` and deprecate `IClock`

### ISSUE THREAD

**Issue by Piedone** [OrchardCMS/OrchardCore#17290]
### Is your feature request related to a problem?

From the 17th December meeting: We now have [`TimeProvider`](https://learn.microsoft.com/en-us/dotnet/api/system.timeprovider) in .NET, which is much of the same as our `IClock` (and more, since it also serves other .NET types). We're now thus duplicating .NET functionality.

Also see [this blog post](https://blog.nimblepros.com/blogs/finally-an-abstraction-for-time-in-net/) and [this one](https://andrewlock.net/exploring-the-dotnet-8-preview-avoiding-flaky-tests-with-timeprovider-and-itimer/).

### Describe the solution you'd like

Use `TimeProvider` and deprecate `IClock`, but keep it in a backward compatible manner.

### Describe alternatives you've considered

We can just keep our parallel implementation, but it's better to align with the wider .NET ecosystem.

**Comment by Piedone** [comment]
Just dropping `IClock` would be hugely breaking. Since it's simple interface, providing a shim will be of little effort, by allowing people to adapt gradually.

**Comment by sebastienros** [comment]
@hishamco I think this was mentioned in the meeting too, that we can keep the `IClock` around for users, that just wraps the new `TimeProvider` implementation, but OC would not use it. And then we can make it obsolete (or not). Not sure this would be removed anytime soon.

**Comment by hishamco** [comment]
Seb do you remember which meeting, I need to check the recording

**Comment by Piedone** [comment]
It's somewhere in this section: https://youtu.be/Cmu7RLimyoQ?si=Hnxv4IMw3t67b85X&t=2073

**Comment by Skrypt** [comment]
I say IClock rulz 😎

**Comment by gvkries** [comment]
> I say IClock rulz 😎

I agree, `TimeProvider` isn't a drop-in replacement for `IClock`. Our implementation of `IClock` offers additional functionality and integrates with Noda Time. That said, we could consider creating a `TimeProvider` implementation that wraps our `IClock`. This would allow developers to use either IClock or TimeProvider interchangeably, depending on their preference or the APIs they’re working with, without having to worry about inconsistencies.
